cypress/e2e/GUI Tests/ui-funcionality-tests.spec.js

```diff
@@ -0,0 +1,149 @@
+describe('GUI Tests', () => {
+
+  beforeEach(() => {
+    cy.visit('/customers');
+    cy.contains('button', 'Accept').click();
+    cy.clearCookies();
+    cy.clearLocalStorage();
+  });
+
+  const selectCompanySize = (size) => {
+    cy.get('[data-testid="size-filter"]').select(size);
+  };
+
+  const clickViewCompanyButton = (companyName) => {
+    cy.get('[data-testid="table"] tbody tr').contains(companyName)
+      .parent()
+      .within(() => {
+        cy.get('button').contains('View').click();
+      });
+  };
+
+  const clickBackButton = () => {
+    cy.get('button').contains('Back').click();
+  }
+
+  const clickFirstViewButton = () => {
+    cy.get('[data-testid="table"] tbody tr')
+      .first()
+      .within(() => {
+        cy.get('button').contains('View').click();
+      });
+  }
+
+  it('Should maintain filters when returning from customer details', () => {
+    selectCompanySize('Small');
+    cy.get('[data-testid="table"] tbody tr').should('have.length', 1).and('contain.text', 'Small');
+    clickViewCompanyButton('Jacobs Co');
+    cy.contains('Customer Details');
+    clickBackButton();
+    cy.get('[data-testid="table"] tbody tr').should('have.length', 1).and('contain.text', 'Small');
+  });
+
+  it('Should return to the customer list after clicking back', () => {
+    clickFirstViewButton();
+    cy.contains('Customer Details');
+    clickBackButton();
+    cy.get('[data-testid="table"] tbody tr').should('have.length', 10)
+  });
+
+  //não consegui pensar em uma maneira simples de fazer esse cara, queria verificar se cada link estava mandando pro lugar correto tbm
+  it('Should display the footer with the correct text and links', () => {
+    cy.get('footer')
+      .should('be.visible')
+      .contains('Copyright 2024 - Talking About Testing');
+  
+    const links = [
+      { text: 'Hotmart', href: 'https://hotmart.com/pt-br/club/cypress-playground-ate-a-nuvem' },
+      { text: 'Udemy', href: 'https://udemy.com/user/walmyr' },
+      { text: 'Blog', href: 'https://talkingabouttesting.com' },
+      { text: 'YouTube', href: 'https://youtube.com/@talkingabouttesting' }
+    ];
+
+    cy.get('footer a')
+      .should('have.length', links.length)
+      .each(($el, index) => {
+        cy.wrap($el).should('have.text', links[index].text);
+        cy.wrap($el).should('have.attr', 'href', links[index].href);
+      });
+  });
+
+  it('Should displays the greeting "Hi, there" when no name is given', () => {
+    cy.get('input[type="text"]').should('have.value', '');
+    cy.get('[data-testid="table"] h2').should('contain.text', 'Hi there!');
+  });
+
+  it('Should displays the greeting "Hi, Joe" when the name Joe is given', () => {
+    cy.get('input[type="text"]').type('Joe');
+    cy.get('[data-testid="table"] h2').should('contain.text', 'Hi Joe!');
+  });
+
+  it('Should display title, theme switcher and input fielda', () => {
+    cy.get('h1').should('contain.text', 'EngageSphere');
```

Comment:
Veja uma solução mais simples:

```suggestion
    cy.contains('h1', 'EngageSphere').should('be.visible');
```

Perceba inclusive a verificação pela visibilidade do elemento.